### PR TITLE
[stdlib] Renames and tests for Swift 3 compatibility mode

### PR DESCRIPTION
**Pull request by moiseev**
Addresses <rdar://problem/32432481>

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please Test Source Compatibility

**Comment by moiseev** [comment]
@jrose-apple This re-introduces the message from #9949. Does this one look better?

File: test/stdlib/IntegerRenames4.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+func absoluteValuable<T : AbsoluteValuable>(x: T) {} // expected-error {{Please use 'SignedNumeric & Comparable' instead.}}
+func _signedInteger<T : _SignedInteger>(x: T) {} // expected-error {{'_SignedInteger' has been renamed to 'SignedInteger'}}
+
+func absolutaValuable<T : SignedNumeric & Comparable>(x: T) {
+  _ = T.abs(x) // expected-error {{use the 'abs(_:)' free function}}
+}
+
+func signedIntegerMaskingArithmetics<T : SignedInteger>(x: T) {
+  _ = x &+ x // expected-error {{use 'FixedWidthInteger' instead of 'SignedInteger' to get '&+' in generic code}}
+  _ = x &- x // expected-error {{use 'FixedWidthInteger' instead of 'SignedInteger' to get '&-' in generic code}}
```
Comment (moiseev) [inline test/stdlib/IntegerRenames4.swift]: @jrose-apple here.

Comment (jrose-apple) [inline test/stdlib/IntegerRenames4.swift]: I don't remember what my original concern was (your link is to this PR, not whatever we talked about before). This looks fine to me now.

**Review by nkcsgexi — APPROVED** [review]
Thank you for adding the rename attributes. LGTM!

File: test/stdlib/IntegerRenames4.swift (hunk 1 above)
Comment (moiseev) [inline test/stdlib/IntegerRenames4.swift]: Argh... This is the right one: https://github.com/apple/swift/pull/9045. The concern was about the message not being very clear.

Comment (jrose-apple) [inline test/stdlib/IntegerRenames4.swift]: Ah, yes, I like the new message a lot better. Thanks, Max!

**Pull request merged**